base/experimental.jl

```diff
@@ -316,15 +316,20 @@ the handler for that type.
 """
 function show_error_hints(io, ex, args...)
     @nospecialize
-    hinters = get(_hint_handlers, typeof(ex), nothing)
-    isnothing(hinters) && return
-    for handler in hinters
-        try
-            @invokelatest handler(io, ex, args...)
-        catch
-            tn = typeof(handler).name
-            @error "Hint-handler $handler for $(typeof(ex)) in $(tn.module) caused an error" exception=current_exceptions()
+    ex_supertype = typeof(ex)
+    while ex_supertype != Any
+        hinters = get(_hint_handlers, ex_supertype, nothing)
+        isnothing(hinters) && continue
+        for handler in hinters
```

Comment:
```suggestion
        for handler in 1:2:length(hinters)
            ex isa hinters[handler] || continue
            handler = hinters[handler + 1]
```

Reply:
I don't understand the point of this one, can you enlighten me?

Reply:
Oh right, I was thrown off by your double use of the `handler` variable. Maybe it would be clearer like this? Or using two zipped lists even?
```suggestion
        for k in 1:2:length(hinters)
            ex isa hinters[k] || continue
            handler = hinters[k + 1]
```